lib/validation/schema.ex

```diff
@@ -7,6 +7,10 @@ defmodule Validation.Schema do
   alias Validation.Result
   alias Validation.Rule
 
+  @type t           :: %__MODULE__{val: schema_fun, meta: meta_data}
+  @typep schema_fun :: ((any) -> Result.t)
```

Comment:
This should be `((Map.t) -> Result.t)`, right? Since the `val` defined in `build` takes a single argument that is set to the `data` of %Result{}, which in turn is defined as `Map.t`